electronics/sections/ultrasound.md

```diff
@@ -19,14 +19,25 @@ The `Robot` library will automatically measure the time between the trigger and
 
 ## Setting up the sensor
 
-To use the sensor, you will need to do some soldering.
+Using the provided PCB, connect the ultrasound sensor to the female header on one side of the board. On the other side of the board, there is a screw terminal. Connect the VCC pin to 5V from a bench power supply and connect the ground pin to the ground connection on the power supply. Connect the trigger to the red wire from the signal generator and the black wire to ground. Then using two channels on the oscilloscope, connect one channel to the trigger pin, and connect a second to the echo pin, connect both their ground connections to the ground of the ultrasound.
 
-\begin{center}  \includegraphics[height=4cm]{img/ultra-strip.png} \end{center}
+[docs]: https://docs.sourcebots.org
 
-1. Solder the female header onto a piece of stripboard.
-2. Solder one length of wire in the same line as each pin. Make sure you use a **red wire** for Vcc and a **black wire** for GND. Other wires should be in any other colour of your choice.
-3. Connect the red wire to 5V on the Arduino and the black wire to GND on the Arduino.
-4. Connect the **Trig** and **Echo** wires to any digital IO lines on the Arduino.
-5. See the [docs][docs] for information on using the sensor.
+## Calculating the speed of sound
 
-[docs]: https://docs.sourcebots.org
+In order to estimate range using your ultrasound, you will need to calculate the speed of sound. Since velocity is given by:
+
+$$ velocity = \frac{distance}{time} $$
+
+You can calculate the speed of sound by taking a known distance, and seeing how long it takes for a response from the ultrasound. 
+
+Set the signal generator to produce a square wave with a frequency of 20Hz, with a low level of 0V, and a high level of 5V. Adjust the duty cycle of your square wave until the 5V period is approximately 10$\mu$s. You can verify this by looking at the value of each square on the oscilloscope's screen.
+
+After 10$\mu$s the ultrasound sensor will then transmit. In order to get an accurate reading, you may need to adjust your oscilloscope's "Trigger" settings. Press the trigger menu button towards the right of the oscilloscope, and set your triggering on the same channel as the trigger pin of your ultrasound sensor, along with setting the coupling to DC and the trigger type to edge. You can also adjust the trigger level, which is the threshold that will cause your oscilloscope to take a reading. See if you can find a section where the echo signal goes high (You may need to adjust your time scale to do this).
+
+\begin{center} \includegraphics[height=8cm]{img/ultrasound-scope.png} \end{center}
+
+You should now see something like the image above. Using the oscilloscope, find the length in seconds of the high section of the echo signal. This is the length of time required to get recieve a signal back. Since the ultrasound signal has to go outward and then return to the sensor, divide this time by 2.
```

Comment:
I think people who aren't experienced with using oscilloscopes or waveforms might struggle to interpret what's on the display, so I think it'd be useful to label which waveform is which.

```suggestion
You should now see something like the image above. Here, the bottom waveform shows the trigger signal and the top waveform shows the echo signal. Using the oscilloscope, find the length in seconds of the high section of the echo signal. This is the length of time required to get recieve a signal back. Since the ultrasound signal has to go outward and then return to the sensor, divide this time by 2.
```